### PR TITLE
Add docs for HTTP/2 Server Push

### PR DESCRIPTION
**Pull request by dunglas**
https://github.com/api-platform/core/pull/2450

### CODE REVIEW

File: core/performance.md [hunk 1]
```diff
@@ -20,11 +20,36 @@ fresh, because the cache is purged in real time.
...
+            enabled: true
+            varnish_urls: ['%env(VARNISH_URL)%']
+        # Adds sensitive default cache headers
+        max_age: 0
+        shared_max_age: 3600
+        vary: ['Content-Type', 'Authorization']
+        public: true
+```
+
+Support for other cache servers than Varnish can easily be added by implementing the `ApiPlatform\Core\HttpCache\PurgerInterface`.
```
Comment (bendavies) [inline core/performance.md]: `Support for cache servers other than Varnish`
or even better
`Support for reverse proxies other than Varnish`

File: core/push-relations.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+>    HTTP/2 allows a server to pre-emptively send (or "push") responses
+     (along with corresponding "promised" requests) to a client in
+     association with a previous client-initiated request.  This can be
+     useful when the server knows the client will need to have those
+     responses available in order to fully process the response to the
+     original request.
+     - [RFC 7540](https://tools.ietf.org/html/rfc7540#section-8.2)
+
+API Platform leverages this capability of the protocol by allowing to push to the clients the relations of a resource:
```
Comment (bendavies) [inline core/push-relations.md]: `API Platform leverages this capability by pushing relations of a resource to clients.`

File: core/push-relations.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+use ApiPlatform\Core\Annotation\ApiProperty;
+use ApiPlatform\Core\Annotation\ApiResource;
+
+/**
+ * @ApiResource
+ */
+class Book
+{
+    /**
+     * @var Author $author
```
Comment (bendavies) [inline core/push-relations.md]: `@var Author`

File: core/push-relations.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+     */
+    public $author;
+}
+```
+
+By setting the `push` attribute to `true` on a property holding a relation, API Platform will automatically add a valid `Link` HTTP header with the `preload` relation.
+According to the [Preload W3C Candidate Recommendation](https://www.w3.org/TR/preload/), web servers and proxy servers can read this header, fetch the related resource and send it to the client using Server Push.
+[NGINX](https://www.nginx.com/blog/nginx-1-13-9-http2-server-push/), [Apache](https://httpd.apache.org/docs/current/howto/http2.html#push), [CloudFlare](https://www.cloudflare.com/website-optimization/http2/serverpush/), [Fastly](https://docs.fastly.com/guides/performance-tuning/http2-server-push) and [Akamai](https://blogs.akamai.com/2017/03/http2-server-push-the-what-how-and-why.html) honor this header. 
+
+Using this feature allows to maximise HTTP cache hits for your API resources.
```
Comment (bendavies) [inline core/push-relations.md]: `Using this feature maximises HTTP cache hits for your API resources.`

File: core/performance.md [hunk 5]
```diff
@@ -20,11 +20,36 @@ fresh, because the cache is purged in real time.
 The support for most specific cases such as the invalidation of collections when a document is added or removed or for
 relationships and inverse relations is built-in.
 
-We also included [Varnish](https://varnish-cache.org/) in the [Docker setup](../distribution/index.md#using-the-official-distribution-recommended) provided with the
-distribution of API Platform, so this feature works out of the box.
+Integration with Varnish and the Doctrine ORM is shipped with the core library, and [Varnish](https://varnish-cache.org/) is included in the [Docker setup](../distribution/index.md#using-the-official-distribution-recommended) provided with the
```
Comment (bendavies) [inline core/performance.md]: `Integration with Varnish and Doctrine ORM`

Comment (dunglas) [reply core/performance.md]: Thanks @bendavies!

**Pull request merged**